tests/communications/test_notification_preferences.py

```diff
@@ -0,0 +1,156 @@
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APITestCase
+from django.contrib.auth import get_user_model
+from rest_framework.authtoken.models import Token
+from communications.models import NotificationType, UserNotificationPreference, UserNotificationTypePreference
+
+import logging
+logging.disable(logging.CRITICAL)
+
+User = get_user_model()
+
+
+class NotificationPreferencesTestCase(APITestCase):
+    def setUp(self):
+        self.user = User.objects.create_user(
+            email='test@example.com',
+            password='testpass123',
+            first_name='Test',
+            last_name='User'
+        )
+        self.token = Token.objects.create(user=self.user)
+        self.client.force_authenticate(user=self.user, token=self.token)
+        
+        self.notification_type1 = NotificationType.objects.create(
+            code='test_type_1',
+            name='Test Type 1',
+            description='Test Description 1',
+            default_frequency='immediate'
+        )
+        self.notification_type2 = NotificationType.objects.create(
+            code='test_type_2',
+            name='Test Type 2',
+            description='Test Description 2',
+            default_frequency='daily_digest'
+        )
+
+    def test_get_notification_types(self):
+        """Test retrieving notification types."""
+        url = reverse('communications:notification-type-list')
+        response = self.client.get(url)
+        
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertIsInstance(response.data, list)
+        codes = [item['code'] for item in response.data]
+        self.assertIn(self.notification_type1.code, codes)
+        self.assertIn(self.notification_type2.code, codes)
+
+    def test_get_user_preferences(self):
+        """Test retrieving user notification preferences."""
+        url = reverse('communications:user-notification-preference-list')
+        response = self.client.get(url)
+        
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertIsInstance(response.data, list)
+        self.assertTrue(len(response.data) > 0)
+        
+        user_prefs = response.data[0]
```

Comment:
user_prefs = response.data[0] assumes the first item is always the correct user. If the queryset is not filtered to the current user in the view, this could test the wrong object.